Makefile

```diff
@@ -101,6 +100,7 @@ clientset: ## Generate a typed clientset
 		--listers-package sigs.k8s.io/cluster-api/pkg/client/listers_generated \
 		--output-package sigs.k8s.io/cluster-api/pkg/client/informers_generated \
 		--go-header-file=./hack/boilerplate.go.txt
+	$(MAKE) gazelle
```

Comment:
Why not keep this with `generate`? Bazel files might need to be updated after `go generate`

Reply:
ok, we need it at both places to make Makefile target idempotent, the ci script `verify_clientset.sh` call make clientset, i will update the PR, thanks

Reply:
I was seeing issues with calling `gazelle` at the end of `clientset` where it was generating the wrong paths in the bazel build files, which was the reason for moving it to the `generate` target.

Reply:
@detiber i replied in another code line thread,  thanks